### ISSUE TITLE
[Feature Request/Question] command line args to R script file not R itself

### ISSUE THREAD

**Issue by TTTPOB** [ManuelHentschel/VSCode-R-Debugger#154]
maybe related #114
I cannot find where I can add command line arguments to the R script file itself rather than R itself, to debug a script file using `argparse`, this feature is needed

**Comment by ManuelHentschel** [comment]
Can you please give an example how the corresponding R/Rscript call would look like?

**Comment by TTTPOB** [comment]
> Can you please give an example how the corresponding R/Rscript call would look like?

sorry for the late reply

it will looks like a python argparse script
`Rscript path/to/the/script --step 1 --dist 100`
inside the script
```r
library(argparse)
parser<-ArgumentParser(description="demo")
attach(parser)
add_argument("--step", "-s", help="step of the counter")
add_argument("--dist", "-d", help="dist of the counter")
detatch(parser)
args<-parser$parse_args()

cat(
  "the dist is: ",
  dist,
  "\n",
  "the step is: ",
  step,
  "\n"
)
```
as you see, the arguments lies after the script, not the `R`/`Rscript`

**Comment by tddschn** [comment]
I need this too!  
Also I'd love to see the option to set `Rscript` args (like `--vanilla`, because my .Rprofile may mess with the debugging or the script).

**Comment by kevinpauli** [comment]
Yes IMO this is required for developing any non-trivial real-world script that needs to be configurable at runtime

**Comment by karchern** [comment]
Are there any updates on this? This would be a very useful feature to have...

**Comment by karchern** [comment]
One can make this work by setting the `--args` flag, followed by whatever command line arguments one wants to supply to the script. One can then use `commandArgs(trailingOnly = TRUE)` to obtain the passed CL arguments. To do this, add a `commandLineArgs` entry into the `launch.json`.

Caveat is that 'proper' command line parsing with e.g. argparse isn't supporte (or at least I don't expect it to work)

![image](https://github.com/ManuelHentschel/VSCode-R-Debugger/assets/19149571/32150a5e-602d-43b3-8edb-3ea9834e58fb)

![image](https://github.com/ManuelHentschel/VSCode-R-Debugger/assets/19149571/c7d5ca60-2fba-4a9c-808a-31b37ffe6528)

**Comment by kevinpauli** [comment]
> argparse isn't supporte (or at least I don't expect it to work)

It actually does work with argparse! 

```
library(argparse)
parser <- ArgumentParser()
parser$add_argument("-n", "--name", help = "Whom shall I greet?", default = "World")
args <- parser$parse_args(commandArgs(trailingOnly = TRUE))
cat(paste0("Hello, ", args$name, "!\n"))
```

```
$ Rscript hello_name.R --name Kevin
Hello, Kevin!
```

![Cursor_and_launch_json_—_corn-plhtn__Dev_Container__corn-plhtn_](https://github.com/ManuelHentschel/VSCode-R-Debugger/assets/2067494/d8d1d109-c17c-4660-9385-d55159c2f069)
![Cursor_and_hello_name_R_—_corn-plhtn__Dev_Container__corn-plhtn_](https://github.com/ManuelHentschel/VSCode-R-Debugger/assets/2067494/3adbd220-dde4-494e-9db2-10dca5a4068c)
